### PR TITLE
Moving WooCommerce class to a separate file and fixing coding standards

### PR DESCRIPTION
**Pull request by claudiosanches**
This is a good start point to fully follow all coding standards and best practices, not to mention to inspire people to do the same, and also will helps after #15798 is merged.

### CODE REVIEW

**Comment by claudiosanches** [comment]
Thanks @aristath to raise this issue.

File: includes/class-woocommerce.php [hunk 1]
```diff
@@ -0,0 +1,655 @@
+<?php
+/**
+ * Main WooCommerce Class
```
Comment (mikejolley) [inline includes/class-woocommerce.php]: Do we need a file docblock which says the same as the class docblock below?

File: woocommerce.php [hunk 2]
```diff
@@ -665,9 +49,9 @@ public function mailer() {
  * @since  2.1
  * @return WooCommerce
  */
-function WC() {
+function wc() {
```
Comment (mikejolley) [inline woocommerce.php]: will this break anything?

File: woocommerce.php [hunk 3]
```diff
@@ -16,646 +16,30 @@
...
-	/**
-	 * Get gateways class.
-	 * @return WC_Payment_Gateways
-	 */
-	public function payment_gateways() {
-		return WC_Payment_Gateways::instance();
-	}
+// Define WC_PLUGIN_FILE.
+if ( ! defined( 'WC_PLUGIN_FILE' ) ) {
+	define( 'WC_PLUGIN_FILE', __FILE__ );
```
Comment (mikejolley) [inline woocommerce.php]: This is the only constant needed here. The rest can use this for __FILE__

**Review by mikejolley — CHANGES_REQUESTED** [review]

File: woocommerce.php (hunk 2 above)
Comment (claudiosanches) [inline woocommerce.php]: PHP functions name are case-insensitive.

See: http://php.net/manual/en/functions.user-defined.php

File: includes/class-woocommerce.php (hunk 1 above)
Comment (claudiosanches) [inline includes/class-woocommerce.php]: We can change what this say, but the best practice is have a docblock for the file and another for the class.

**Pull request merged**